tldw_Server_API/app/core/AuthNZ/api_key_audit.py

```
@@ -0,0 +1,82 @@
+        await audit_service.flush(raise_on_failure=True)
+    except MandatoryAuditWriteError:
+        raise
+    except Exception as exc:
+        logger.opt(exception=True).error(
+            "Mandatory API key management audit write failed (action={}, user_id={}, resource_id={}): {}",
+            action,
+            user_id,
+            resource_id,
+            type(exc).__name__,
+        )
+        raise MandatoryAuditWriteError("Mandatory audit persistence unavailable") from exc
```

Ruff (0.15.9)

[warning] Avoid specifying long messages outside the exception class

(TRY003)